src/main/scala/mesosphere/marathon/core/storage/MarathonState.scala

```diff
@@ -0,0 +1,22 @@
+package mesosphere.marathon.core.storage
+
+import akka.util.ByteString
+import com.google.protobuf.{ MessageLite, Parser }
+import mesosphere.marathon.state.Timestamp
+
+trait MarathonState[Proto <: MessageLite] {
```

Comment:
Do we really need this? It means that all entity types have to implement this.
I like the idea more to have a type class implement the proto handling.

I guess everything we need is to add this into  MarathonProto:
```scala
trait MarathonProto[Proto <: MessageLite, T] {
  def toProto(t: T): Proto
  def version(t: T): Timestamp
  ...
```